### PR TITLE
Client changes for deleting outcome

### PR DESCRIPTION
**Pull request by johngonz0**
client patch for deleting outcomes

https://github.com/Cyber4All/clark-client/assets/150100412/b53331b2-7e17-4a40-aa77-79d91a735a3a

### CODE REVIEW

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 1]
```diff
@@ -518,17 +518,19 @@ export class BuilderStore {
...
 
   private deleteOutcome(id: string) {
+    console.log(id);
+    console.log('in builder delete outcome');
     // grab the outcome that's about to be deleted
     const outcome: Partial<LearningOutcome> = this.outcomes.get(id);
-
     // delete the outcome
-    this.outcomes.delete(id);
+    this.outcomes.delete(outcome.serviceId);
```
Comment (johngonz0) [inline src/app/onion/learning-object-builder/builder-store.service.ts:526]:
```suggestion
    this.outcomes.delete(id);
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 2]
```diff
@@ -574,7 +573,7 @@ export class BuilderStore {
       },
       true
     );
-
+    console.log(outcome);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:576]:
```suggestion
```
🔥

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 3]
```diff
@@ -1061,6 +1061,7 @@ export class BuilderStore {
       // clear the cache here so that new requests start a new cache
       this.outcomeCache$.next(undefined);
 
+      console.log(newValue);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:1064]:
```suggestion
```
🔥

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 4]
```diff
@@ -1098,6 +1099,7 @@ export class BuilderStore {
     this.outcomeService
       .addLearningOutcome(this.learningObject._id, newOutcome)
       .then((serviceId: string) => {
+        console.log(serviceId);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:1102]:
```suggestion
```
🔥

File: src/app/onion/learning-object-builder/components/outcome/outcome.component.ts [hunk 5]
```diff
@@ -111,6 +111,7 @@ export class OutcomeComponent implements OnInit {
   }
 
   emitDeletion() {
+    console.log('emitting deletion');
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/components/outcome/outcome.component.ts:114]:
```suggestion
```
🔥

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 6]
```diff
@@ -1107,6 +1109,7 @@ export class BuilderStore {
         } = this.outcomes.get(newOutcome.id);
         // store the temporary id in the outcome so that the page component know's which outcome to keep focused
         outcome.serviceId = serviceId;
+        console.log(outcome.serviceId);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/builder-store.service.ts:1112]:
```suggestion
```
🇧🇫

File: src/entity/learning-outcome/learning-outcome.ts [hunk 7]
```diff
@@ -21,6 +21,14 @@ export class LearningOutcome {
...
   }
 
+  private _serviceId!: string;
+  get serviceId(): string {
+    return this.serviceId;
+  }
+  set serviceId(serviceId: string) {
+    this._serviceId = serviceId;
+  }
+
```
Comment (mfranz1) [inline src/entity/learning-outcome/learning-outcome.ts:31]: Could these get a few inline or bloc comments?

**Review by mfranz1 — CHANGES_REQUESTED** [review]
yeah no

**Review by curlyLasagna — APPROVED** [review]

File: src/app/onion/core/learning-object.service.ts [hunk 8]
```diff
@@ -284,8 +284,9 @@ export class LearningObjectService {
    * @param outcomeId The outcome Id
    */
   deleteOutcome(outcomeId: string): Promise<any> {
+    console.log('here');
```
Comment (johngonz0) [inline src/app/onion/core/learning-object.service.ts:287]:
```suggestion
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 9]
```diff
@@ -547,7 +546,7 @@ export class BuilderStore {
     params: { verb?: string | undefined; bloom?: string | undefined; text?: string }
   ) {
     const outcome = this.outcomes.get(id);
-
+    console.log(outcome);
```
Comment (johngonz0) [inline src/app/onion/learning-object-builder/builder-store.service.ts:549]:
```suggestion
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 10]
```diff
@@ -1061,6 +1059,7 @@ export class BuilderStore {
       // clear the cache here so that new requests start a new cache
       this.outcomeCache$.next(undefined);
 
+      console.log(newValue);
```
Comment (johngonz0) [inline src/app/onion/learning-object-builder/builder-store.service.ts:1062]:
```suggestion
```

**Review by stef359 — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

**Pull request merged**